### ISSUE TITLE
`ecs-cli compose service up` doesn't do an upgrade unless the docker-compose.yml has changed

### ISSUE THREAD

**Issue by kaihendry** [aws/amazon-ecs-cli#144]
```
$ ecs-cli compose service up
INFO[0001] Using ECS task definition                     TaskDefinition=ecscompose-count:4
INFO[0001] ECS Service is already running                desiredCount=1 serviceName=ecscompose-service-count
INFO[0001] ECS Service has reached a stable state        desiredCount=1 runningCount=1 serviceName=ecscompose-service-count
```

I'm told `ecs-cli compose service up` is used to roll out an update but you get a **ECS Service is already running** if you don't change the image: url in the `docker-compose.yml`. I expected `ecs-cli` to detect my dkr image on that URL is newer than the one running!!

My workaround is to add and remove the `:latest` tag to trigger an update. But this is manual and we need this to be automatic.

Once it's changed, only then you will get output like:

```
$ ecs-cli compose service up
INFO[0001] Using ECS task definition                     TaskDefinition=ecscompose-count:3
INFO[0001] Updated the ECS service with a new task definition. Old containers will be stopped automatically, and replaced with new ones  desiredCount=2 serviceName=ecscompose-service-count taskDefinition=ecscompose-count:3
INFO[0001] Describe ECS Service status                   desiredCount=2 runningCount=2 serviceName=ecscompose-service-count
```

**Comment by tallavi** [comment]
Yes, I'm having the same issue. Usually I'm doing a stop and then a start (I have a script for that), or for services, I just stop the task and let the service create a new task.

**Comment by panga** [comment]
+1

**Comment by hlesesne** [comment]
I came up with a solution (in the form of a build.sh) that simply timestamps the tag and alters the docker-compose.yml then pushes.

```
cd ./backend
docker build -t pc-issues .

ts=`date +"%Y%m%d%H%M%S"`
tag="565611716335.dkr.ecr.us-east-1.amazonaws.com/pc-issues:$ts"
orig="##TAG##"
echo "TAG $tag"
sed "s~$orig~$tag~g" "docker-compose.template.yml" > "docker-compose.yml"
docker tag pc-issues:latest $tag
docker push $tag
ecs-cli compose --file docker-compose.yml up
```


My docker-compose.template.yml looks kinda like this:
```
version: '2'
services:
    appserver:
        image: ##TAG##
        cpu_shares: 100
        mem_limit: 524288000
        ports:
            - "3030:3030"
```

resulting in a docker-compose.yml file kinda like this with each run:
```
version: '2'
services:
    appserver:
        image: 10101010101010101.dkr.ecr.us-east-1.amazonaws.com/pc-issues:20161208214812
        cpu_shares: 100
        mem_limit: 524288000
        ports:
            - "3030:3030"
```

YMMV

**Comment by markriggins** [comment]
I highly recommend using SEMVER-style GIT tags and the `git describe` command for tagging docker images. That way you can tell more about an image, what its tag actually means semantically and when it was created.  If there have been any commits since the last tag was created, the `git describe` command will add a suffix in the format -N-g<sha>  with N being the number of commits since the tag, and the <sha> being the short GIT hash of the latest commit.

**Comment by allisaurus** [comment]
Hi all - we recently merged in support for the new "--force-deployment" option, which will allow you to force a new service deployment on "compose service up" without having to edit your task definition. It should be available in our next release. Thanks for your feedback re: this issue!

**Comment by AnkurJain10** [comment]
@allisaurus Can you please provide the release date and version number for your next release that will contain the --force-deployment tag.

**Comment by allisaurus** [comment]
@AnkurJain10  We don't have those specifics to share yet, but if you'd like to use it now you can build from source (using the "dev" branch)

**Comment by alihalabyah** [comment]
@allisaurus Was the --force-deployment released?

**Comment by allisaurus** [comment]
Hi @alihalabyah : It's not been released yet; but we're currently planning our next release and we'll be sure to update this issue when we have more details!

**Comment by allisaurus** [comment]
@alihalabyah & others:  the --force-deployment feature has been released in v1.4.0 is now available for download. Thanks for your patience while we pulled this across the finish line.

**Comment by haydenth** [comment]
Thank you for doing this!! Saved me so much work of trying to hack this :)
